File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/Queries/GetOrderByIdQuery.cs
using MediatR;
using MediatorDesignPattern.Example2.Models;

namespace MediatorDesignPattern.Example2.Queries;

/// <summary>
/// MediatR Query - Query order by ID
/// </summary>
public class GetOrderByIdQuery : IRequest<Order?>
{
    public Guid OrderId { get; set; }
}

File: java/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/src/main/java/behavioral/mediator/example2/GetOrderByIdQuery.java
package behavioral.mediator.example2;

import java.util.UUID;

public class GetOrderByIdQuery implements IRequest<Order> {
    private final UUID orderId;
    public GetOrderByIdQuery(UUID orderId) { this.orderId = orderId; }
    public UUID getOrderId() { return orderId; }
}
